### PR TITLE
Improve error message and alert display

### PR DESCRIPTION
**Pull request by tomasbrod**
Improve error message display.
Display Alert protocol messages.
Warn if difficulty < 0.1.
Change Alert **key** to be same as project **master** key.

In future the messages should be changed to enums and have GUI determine the text to display.

### CODE REVIEW

File: src/miner.cpp [hunk 1]
```diff
@@ -826,7 +815,6 @@ void StakeMiner(CWallet *pwallet)
         MiningCPID BoincData;
         { LOCK(MinerStatus.lock);
             //clear miner messages
-            MinerStatus.Message="";
```
Comment (tomasbrod) [inline src/miner.cpp:829]: error here. not clearing message will result in runaway allocation. either clear it or remove Message from struct completely.

File: src/alert.cpp [hunk 2]
```diff
@@ -250,8 +251,8 @@ bool CAlert::ProcessAlert(bool fThread)
                 // the whole string before passing it to the shell:
                 std::string singleQuote("'");
                 // safeChars chosen to allow simple messages/URLs/email addresses, but avoid anything
-                // even possibly remotely dangerous like & or >
-                std::string safeChars("abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ01234567890 .,;_/:?@");
+                // even possibly remotely dangerous like & ; $ or >
+                std::string safeChars("abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ01234567890 .,_/:?@");
```
Comment (denravonska) [inline src/alert.cpp:255]: Is this stored in the appcache later? This should be much safer with the appcache map-map changes, right?

Comment (tomasbrod) [reply src/alert.cpp:255]: No, the safened string is passed as a command-line argument to a user specified command. I removed a semicolon character from the set of safe characters as it is often used for separating commands. The string is enclosed in single quotes later, so yeah, the semicolon can't do much there. Extra caution does not hurt I guess.

**Comment by tomasbrod** [comment]
What is holding back this PR?

**Comment by denravonska** [comment]
Me who keeps assuming there was an open question when I read the title. I'll try to get to it this evening.

File: src/alert.cpp [hunk 3]
```diff
@@ -20,7 +20,8 @@ using namespace std;
 map<uint256, CAlert> mapAlerts;
 CCriticalSection cs_mapAlerts;
 
-static const char* pszMainKey = "0486bce1bac0d543f104cbff2bd23680056a3b9ea05e1137d2ff90eeb5e08472eb500322593a2cb06fbf8297d7beb6cd30cb90f98153b5b7cce1493749e41e0284";
+// same as master project key now
+static const char* pszMainKey = "049ac003b3318d9fe28b2830f6a95a2624ce2a69fb0c0c7ac0b513efcc1e93a6a6e8eba84481155dd82f2f1104e0ff62c69d662b0094639b7106abc5d84f948c0a";
```
Comment (denravonska) [inline src/alert.cpp:24]: Why did we change this one?

File: src/main.cpp [hunk 4]
```diff
@@ -517,9 +517,35 @@ void GetGlobalStatus()
...
+
+        GlobalStatusStruct.status.clear();
+
+        if(MinerStatus.WeightSum)
+            GlobalStatusStruct.coinWeight = RoundToString(MinerStatus.WeightSum / 80.0,2);
+
+        GlobalStatusStruct.errors.clear();
+        std::string Alerts = GetWarnings("statusbar");
+        if(!Alerts.empty())
+            GlobalStatusStruct.errors += "Alert: " + Alerts + "; ";
```
Comment (denravonska) [inline src/main.cpp]: Wrap these strings (the textual ones) in `_()` macro so they're translatable.

**Comment by denravonska** [comment]
Looks great. Just add the translation macros and we're good to go.

File: src/alert.cpp (hunk 3 above)
Comment (tomasbrod) [reply src/alert.cpp:24]: > Change Alert **key** to be same as project **master** key.

I suspect that secret key to this key was lost as this feature was never used. I made it same as project master key so Quez and RTM can use it.

File: src/miner.cpp [hunk 5]
```diff
@@ -452,7 +451,7 @@ bool CreateCoinStake( CBlock &blocknew, CKey &key,
         || !wallet.SelectCoinsForStaking(BalanceToStake*2, txnew.nTime, CoinsToStake, nValueIn) )
     {
         LOCK(MinerStatus.lock);
-        MinerStatus.ReasonNotStaking+="No coins to stake; ";
+        MinerStatus.ReasonNotStaking+="No coins";
```
Comment (denravonska) [inline src/miner.cpp]: These ones will need macros as well.

**Comment by tomasbrod** [comment]
I added the macros and changed separators in the messages as well.

**Pull request merged**